src/main/scala/com/raphtory/core/components/querymanager/QueryManager.scala

```diff
@@ -150,6 +156,7 @@ class QueryManager(scheduler: Scheduler, conf: Config, pulsarController: PulsarC
       if (safe) maxTime else minTime
     }
     else 0 // not received a message from each partition yet
+    QueryTelemetry.totalQueryManagerWatermarks.inc()
```

Comment:
I don't think this one is needed. Would be a good idea to record the global watermark though